File: node/lambda/src/handlers/CancelAndStopIntentHandler.ts
import { HandlerInput, RequestHandler } from "ask-sdk-core";
import { Response } from "ask-sdk-model";
import { APP_ID } from "..";

export const CancelAndStopIntentHandler: RequestHandler = {
  canHandle(handlerInput: HandlerInput): boolean {
    const request = handlerInput.requestEnvelope.request;
    return (
      request.type === "IntentRequest" &&
      (request.intent.name === "AMAZON.CancelIntent" ||
        request.intent.name === "AMAZON.StopIntent")
    );
  },
  handle(handlerInput: HandlerInput): Response {
    const speakOutput = "Goodbye!";
    return handlerInput.responseBuilder
      .speak(speakOutput)
      .withSimpleCard(APP_ID, speakOutput)
      .getResponse();
  },
};

File: java/lambda/src/com/github/dannful/AniLexa/handlers/CancelandStopIntentHandler.java
package com.github.dannful.AniLexa.handlers;

import static com.amazon.ask.request.Predicates.intentName;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;
import com.github.dannful.AniLexa.Constants;
import java.util.Optional;

public class CancelandStopIntentHandler implements RequestHandler {

  @Override
  public boolean canHandle(HandlerInput input) {
    return input.matches(
      intentName("AMAZON.StopIntent").or(intentName("AMAZON.CancelIntent"))
    );
  }

  @Override
  public Optional<Response> handle(HandlerInput input) {
    String speechText = "Goodbye";
    return input
      .getResponseBuilder()
      .withSpeech(speechText)
      .withSimpleCard(Constants.APP_ID, speechText)
      .build();
  }
}
